### ISSUE TITLE
Parsing error for the dice grammar test cases

### ISSUE THREAD

**Issue by julianthome** [antlr/grammars-v4#843]
`maven-test-plugin` and `inmemantlr` fail to parse the dice test cases. `maven-test-plugin` produce the following output:

```
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/big_d.txt
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/big_dice.txt
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/dice_arithmetic.txt
Parse tree for 'dice_arithmetic.txt' matches 'dice_arithmetic.txt.tree'
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/dice_with_sub.txt
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/dice_with_sum.txt
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/invalid_space_end.txt
line 1:2 token recognition error at: ' '
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/invalid_space_start.txt
line 1:1 token recognition error at: ' '
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/missing_dice.txt
line 1:0 extraneous input 'd' expecting DIGIT
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/missing_side.txt
line 2:0 no viable alternative at input '1d'
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/mixed_arithmetic.txt
Parse tree for 'mixed_arithmetic.txt' matches 'mixed_arithmetic.txt.tree'
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/negative_dice.txt
line 1:0 extraneous input '-' expecting DIGIT
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/negative_number.txt
line 1:0 extraneous input '-' expecting DIGIT
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/no_dice.txt
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/num_arithmetic.txt
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/number.txt
Parsing :/Users/julian/Repositories/inmemantlr/inmemantlr-api/src/test/resources/grammars-v4/dice/examples/simple_dice.txt
```

**Comment by teverett** [comment]
There are .errors files for these.  I believe what that author has done in this case is intentionally provide examples that should not parse, and accompanying .errors files.

**Comment by Bernardo-MG** [comment]
Yes, I tried to add as many cases as possible to the grammar. Before that I was testing the generated files directly, so I kept a broad scope for them.

In the case of the error tests, I tried to cover a few ambiguous cases. The two tests which verify that dice with empty spaces thrown an error were added because the grammar used to have that problem, spaces were ignored in all cases and invalid notation could be parsed as valid.
